iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java

```diff
@@ -2053,7 +2053,7 @@ private void closeUnsealedTsFileProcessorCallBack(TsFileProcessor tsFileProcesso
     closeQueryLock.writeLock().lock();
     try {
       tsFileProcessor.close();
-      if (tsFileProcessor.isEmpty()) {
+      if (tsFileProcessor.isEmpty() || tsFileProcessor.getTsFileResource().isEmpty()) {
```

Comment:
hi , tsfile exsit and resource does not exsit,  will delete tsfile?

Reply:
In this case, tsfile contains no data and resource file exists. 
I'll update the PR description later.

Reply:
ok